pytorch_lightning/callbacks/base.py

```diff
@@ -178,3 +178,19 @@ def on_before_zero_grad(self, trainer, pl_module, optimizer):
         Called after optimizer.step() and before optimizer.zero_grad().
         """
         pass
+
+    def log(self, *_, **__):
+        """
+        This function is a placeholder. Don't override this function.
+        When using Lightning Trainer, the log function from
+        the model will be attached to this callback.
+        """
+        pass
```

Comment:
Why does one method `pass` and the other `raise NotImplementedError`?

Reply:
Done !